docs/recipes/atom-creators.mdx

```diff
@@ -1,521 +0,0 @@
-      </button>
-      <button onClick={() => toggle(true)}>force true</button>
-      <button onClick={() => toggle(false)}>force false</button>
-    </>
-  )
-}
-```
-
-## atomWithToggleAndStorage
-
-> `atomWithToggleAndStorage` is like `atomWithToggle` but also persist the state anytime it changes in given storage using [`atomWithStorage`](../utilities/storage.mdx).
-
-Here is the source:
-
-```ts
-import { WritableAtom, atom } from 'jotai'
-import { atomWithStorage } from 'jotai/utils'
-
-export function atomWithToggleAndStorage(
-  key: string,
-  initialValue?: boolean,
-  storage?: any
-): WritableAtom<boolean, boolean | undefined> {
-  const anAtom = atomWithStorage(key, initialValue, storage)
-  const derivedAtom = atom(
-    (get) => get(anAtom),
-    (get, set, nextValue?: boolean) => {
-      const update = nextValue ?? !get(anAtom)
-      set(anAtom, update)
-    }
-  )
-
-  return derivedAtom
-}
-```
-
-And how it's used:
-
-```js
-import { atomWithToggleAndStorage } from 'XXX'
-
-// will have an initial value set to false & get stored in localStorage under the key "isActive"
-const isActiveAtom = atomWithToggleAndStorage('isActive')
-```
-
-The usage in a component is also the same as `atomWithToggle`.
-
-## atomWithCompare
-
-> `atomWithCompare` creates atom that triggers updates when custom compare function `areEqual(prev, next)` is false.
-
-This can help you avoid unwanted re-renders by ignoring state changes that don't matter to your application.
-
-Note: Jotai uses `Object.is` internally to compare values when changes occur. If `areEqual(a, b)` returns false, but `Object.is(a, b)` returns true, Jotai will not trigger an update.
-
-```ts
-import { atomWithReducer } from 'jotai/utils'
-
-export function atomWithCompare<Value>(
-  initialValue: Value,
-  areEqual: (prev: Value, next: Value) => boolean
-) {
-  return atomWithReducer(initialValue, (prev: Value, next: Value) => {
-    if (areEqual(prev, next)) {
-      return prev
-    }
-
-    return next
-  })
-}
-```
-
-Here's how you'd use it to make an atom that ignores updates that are shallow-equal:
-
-```ts
-import { atomWithCompare } from 'XXX'
-import { shallowEquals } from 'YYY'
-import { CSSProperties } from 'react'
-
-const styleAtom = atomWithCompare<CSSProperties>(
-  { backgroundColor: 'blue' },
-  shallowEquals
-)
-```
-
-In a component:
-
-```jsx
-const StylePreview = () => {
-  const [styles, setStyles] = useAtom(styleAtom)
-
-  return (
-    <div>
-      <div styles={styles}>Style preview</div>
-
-      {/* Clicking this button twice will only trigger one render */}
-      <button onClick={() => setStyles({ ...styles, backgroundColor: 'red' })}>
-        Set background to red
-      </button>
-
-      {/* Clicking this button twice will only trigger one render */}
-      <button onClick={() => setStyles({ ...styles, fontSize: 32 })}>
-        Enlarge font
-      </button>
-    </div>
-  )
-}
-```
-
-## atomWithRefresh
```

Comment:
@gerzonc It seems like this section is lost. Can you double check please?

Reply:
True OMG 🥺 I'll add it right away!